src/components/results/sensitivityAnalysis/paged-sensitivity-analysis-result.js

```diff
@@ -0,0 +1,184 @@
+
+    const filtersDef = useMemo(() => {
+        const baseFilters = [
+            {
+                field: 'funcId',
+                label: intl.formatMessage({
+                    id: sensiKindIndex < 2 ? 'SupervisedBranches' : 'BusBarBus',
+                }),
+                options: result?.allFunctionIds || [],
+            },
+            {
+                field: 'varId',
+                label: intl.formatMessage({ id: 'VariablesToSimulate' }),
+                options: result?.allVariableIds || [],
+            },
+        ];
+
+        if (nOrNkIndex > 0) {
+            baseFilters.push({
+                field: 'contingencyId',
+                label: intl.formatMessage({ id: 'ContingencyId' }),
+                options: result?.allContingencyIds || [],
+            });
+        }
+
+        return baseFilters;
+    }, [intl, sensiKindIndex, nOrNkIndex, result]);
+
+    const { updateFilter, getFilterSelector } = useRowFilter(
+        DATA_KEY_TO_FILTER_KEY
+    );
+
+    // Add default sort on sensitivity col
+    const colKey = 'SENSITIVITY';
+    const sortWay = SORT_WAYS.desc;
+    const { onSortChanged, sortSelector } = useAgGridSort(
+        DATA_KEY_TO_SORT_KEY,
+        { colKey: colKey, sortWay }
+    );
+
+    const handleChangePage = useCallback((_, newPage) => {
+        setPage(newPage);
+    }, []);
+
+    const handleChangeRowsPerPage = useCallback((event) => {
+        setRowsPerPage(parseInt(event.target.value, 10));
+        setPage(0);
+    }, []);
+
+    const fetchResult = useCallback(() => {
+        setIsLoading(true);
+
+        const selector = {
+            isJustBefore: !nOrNkIndex,
+            functionType: FUNCTION_TYPES[sensiKindIndex],
+            offset: page * rowsPerPage,
+            chunkSize: rowsPerPage,
+            ...getFilterSelector(),
+            ...sortSelector,
+        };
+
+        setAllowsShowingLoader(false);
+        timerRef.current = setTimeout(() => {
+            setAllowsShowingLoader(true);
+        }, LOADER_TIMEOUT_MS);
+
+        fetchSensitivityAnalysisResult(studyUuid, nodeUuid, selector)
+            .then((res) => {
+                setResult(res);
+                clearTimeout(timerRef.current);
+            })
+            .catch((err) => {
+                console.warn('sensitivity analysis results fetch', err.message);
+            })
+            .finally(() => {
+                setIsLoading(false);
+            });
+    }, [
+        nOrNkIndex,
+        sensiKindIndex,
+        studyUuid,
+        nodeUuid,
+        page,
+        rowsPerPage,
+        getFilterSelector,
+        sortSelector,
+    ]);
+
+    useEffect(() => {
+        fetchResult();
+        return () => {
+            clearTimeout(timerRef.current);
+        };
+    }, [fetchResult]);
+
+    useEffect(() => {
+        if (result) {
+            const { filteredSensitivitiesCount = 0 } = result;
+            setCount(filteredSensitivitiesCount);
+        }
+    }, [result]);
```

Comment:
set count as soon as you set the result instead of setting a useEffect ?

Reply:
done, thank you